contracts/src/river/registry/facets/stream/StreamRegistry.sol

```diff
@@ -236,15 +236,15 @@ contract StreamRegistry is IStreamRegistry, RegistryModifiers {
   ) external view returns (StreamWithId[] memory, bool) {
     require(start < stop, RiverRegistryErrors.BAD_ARG);
 
-    StreamWithId[] memory streams = new StreamWithId[](stop - start);
-
     uint256 streamCount = ds.streams.length();
 
-    for (
-      uint256 i = 0;
-      ((start + i) < streamCount) && ((start + i) < stop);
-      ++i
-    ) {
+    uint256 maxStreamIndex = stop;
+    if (streamCount < stop) {
+      maxStreamIndex = streamCount;
+    }
+    StreamWithId[] memory streams = new StreamWithId[](maxStreamIndex - start);
```

Comment:
@giuseppecrj @shuhuiluo cant this have been done without a copy using assembly

Reply:
Return data has to be copied to memory.